lib/posix/Kconfig.sched

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config POSIX_PRIORITY_SCHEDULING
```

Comment:
Would it be ok if this name were more along the lines of `POSIX_SCHED`?

The main reason is because the `_POSIX_PRIORITY_SCHEDULING` is a feature test macro that should only be defined when certain features are present.

Some more info here 
https://linux.die.net/man/7/posixoptions

The goal was to tie this together with Kconfig and sysconf and there is even a PR that's getting us some of the way there, which is nice.

https://github.com/zephyrproject-rtos/zephyr/pull/67500

Reply:
Sorry, please feel free to ignore.